docs/source/conf.py

```diff
@@ -126,25 +126,23 @@ def render_autodoc_modules():
 
 import recommonmark
 from recommonmark.transform import AutoStructify
+from recommonmark.parser import CommonMarkParser
 
 
 def setup(app):
     app.add_config_value('recommonmark_config', {'enable_eval_rst': True}, True)
     app.add_stylesheet('custom.css')
     app.add_transform(AutoStructify)
-
-
-source_parsers = {
-   '.md': 'recommonmark.parser.CommonMarkParser',
-}
-source_suffix = ['.rst', '.md']
+    app.add_source_parser(CommonMarkParser)
+    app.add_source_suffix('.md', 'markdown')
+    app.add_source_suffix('.rst', 'restructuredtext')
```

Comment:
@missingcharacter, I'm not familiar with Spinx... Could you please explain a bit why is it needed to move the source_parsers and the source_suffix to the setup stage?

Reply:
@GeorgianaElena it may not be necessary, but I got these warnings:

```
/path/to/venv/lib/python3.8/site-packages/sphinx/util/compat.py:30: RemovedInSphinx30Warning: The config variable "source_parsers" is deprecated. Please update your extension for the parser and remove the setting.
  warnings.warn('The config variable "source_parsers" is deprecated. '
/path/to/venv/lib/python3.8/site-packages/sphinx/util/compat.py:36: RemovedInSphinx30Warning: app.add_source_parser() does not support suffix argument. Use app.add_source_suffix() instead.
  app.add_source_parser(suffix, parser)
```

and decided it may be better to update these too

Reply:
for more details https://www.sphinx-doc.org/en/master/extdev/appapi.html#sphinx.application.sphinx.add_source_parser and https://www.sphinx-doc.org/en/master/extdev/appapi.html#sphinx.application.sphinx.add_source_suffix

Reply:
the old way sphinx would link particular parsers to a specific file extension was the `source_parsers = ` approach...I believe that now extensions automatically tell Sphinx what kind of files they parse, so it's not necessary unless we explicitly need to choose a parser.

Reply:
@choldgraf are you saying it is not necessary to declare:

```python
    app.add_source_parser(CommonMarkParser)
```

or 

```python
    app.add_source_suffix('.md', 'markdown')
    app.add_source_suffix('.rst', 'restructuredtext')
```

or neither?

Reply:
I *think* that both should be automatically figured out if the right extensions are loaded, so no they don't need to be explicitly added (assuming that we support a minimum version of sphinx that supports the feature, though I think this is old-ish). Instead of adding the CommonMark parser explicitly, I think we just need to add `recommonmark` to the list of extensions:

https://github.com/jupyterhub/oauthenticator/pull/345/files#diff-1ec61223727a10e5e4c74c68336809fcR115

instructions from their docs: https://recommonmark.readthedocs.io/en/latest/

Reply:
@choldgraf seems like it worked

```shell
 $ make html
Running Sphinx v2.4.4
Autogenerating module documentation in /path/to/oauthenticator/docs/source/api/gen/oauthenticator.oauth2.rst with classes: {'classes': ['OAuthLoginHandler', 'OAuthCallbackHandler'], 'configurables': ['OAuthenticator']}
Autogenerating module documentation in /path/to/oauthenticator/docs/source/api/gen/oauthenticator.auth0.rst with classes: defaultdict(<class 'list'>, {'configurables': ['Auth0OAuthenticator', 'LocalAuth0OAuthenticator']})
Autogenerating module documentation in /path/to/oauthenticator/docs/source/api/gen/oauthenticator.azuread.rst with classes: defaultdict(<class 'list'>, {'configurables': ['AzureAdOAuthenticator', 'LocalAzureAdOAuthenticator']})
Autogenerating module documentation in /path/to/oauthenticator/docs/source/api/gen/oauthenticator.bitbucket.rst with classes: defaultdict(<class 'list'>, {'configurables': ['BitbucketOAuthenticator', 'LocalBitbucketOAuthenticator']})
Autogenerating module documentation in /path/to/oauthenticator/docs/source/api/gen/oauthenticator.cilogon.rst with classes: defaultdict(<class 'list'>, {'configurables': ['CILogonOAuthenticator', 'LocalCILogonOAuthenticator']})
Autogenerating module documentation in /path/to/oauthenticator/docs/source/api/gen/oauthenticator.generic.rst with classes: defaultdict(<class 'list'>, {'configurables': ['GenericOAuthenticator', 'LocalGenericOAuthenticator']})
Autogenerating module documentation in /path/to/oauthenticator/docs/source/api/gen/oauthenticator.github.rst with classes: defaultdict(<class 'list'>, {'configurables': ['GitHubOAuthenticator', 'LocalGitHubOAuthenticator']})
Autogenerating module documentation in /path/to/oauthenticator/docs/source/api/gen/oauthenticator.gitlab.rst with classes: defaultdict(<class 'list'>, {'configurables': ['GitLabOAuthenticator', 'LocalGitLabOAuthenticator']})
Autogenerating module documentation in /path/to/oauthenticator/docs/source/api/gen/oauthenticator.globus.rst with classes: defaultdict(<class 'list'>, {'configurables': ['GlobusOAuthenticator', 'LocalGlobusOAuthenticator']})
Autogenerating module documentation in /path/to/oauthenticator/docs/source/api/gen/oauthenticator.google.rst with classes: defaultdict(<class 'list'>, {'configurables': ['GoogleOAuthenticator', 'LocalGoogleOAuthenticator']})
Autogenerating module documentation in /path/to/oauthenticator/docs/source/api/gen/oauthenticator.okpy.rst with classes: defaultdict(<class 'list'>, {'configurables': ['LocalOkpyOAuthenticator', 'OkpyOAuthenticator']})
Autogenerating module documentation in /path/to/oauthenticator/docs/source/api/gen/oauthenticator.openshift.rst with classes: defaultdict(<class 'list'>, {'configurables': ['LocalOpenShiftOAuthenticator', 'OpenShiftOAuthenticator']})
Autogenerating module documentation in /path/to/oauthenticator/docs/source/api/gen/oauthenticator.mediawiki.rst with classes: defaultdict(<class 'list'>, {'configurables': ['MWOAuthenticator']})
Adding copy buttons to code blocks...
WARNING: html_static_path entry '_static' does not exist
loading pickled environment... done
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 15 source files that are out of date
updating environment: 0 added, 15 changed, 0 removed
reading sources... [100%] index
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] index
generating indices...  genindex py-modindexdone
writing additional pages...  searchdone
copying static files... ... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded, 1 warning.

The HTML pages are in build/html.
```